### ISSUE TITLE
False positive on implement_methods in complex inheritance setup

### ISSUE THREAD

**Issue by fdupret** [abaplint/abaplint#3716]
Interfaces:
```
lif_animal
  └── make_sound( )

lif_pet
  └── inherits lif_animal
```
Classes:
```
lcl_animal (ABSTRACT)
  └── implements lif_animal
        └── make_sound( ) ABSTRACT

lcl_pet (ABSTRACT)
  └── inherits from lcl_animal
  └── implements lif_pet

lcl_dog
  └── inherits from lcl_pet
  └── redefines lif_animal~make_sound
```
Expected behavior: no error. No implementation is required for `lcl_pet` because it is abstract and `lcl_animal` already declared `make_sound` as abstract. 
Actual behavior: `lcl_pet` returns an error `Implement method "make_sound" from interface "lif_animal"(implement_methods)`.

Minimal reproducible example:
```abap
INTERFACE lif_animal.
  METHODS make_sound.
ENDINTERFACE.


INTERFACE lif_pet.
  " We don't have the issue if the next statement is commented out
  INTERFACES lif_animal.
ENDINTERFACE.


CLASS lcl_animal DEFINITION ABSTRACT.
  PUBLIC SECTION.
    INTERFACES lif_animal
      ABSTRACT METHODS make_sound.

ENDCLASS.


CLASS lcl_animal IMPLEMENTATION.
ENDCLASS.


CLASS lcl_pet DEFINITION
  INHERITING FROM lcl_animal ABSTRACT.

  PUBLIC SECTION.
    INTERFACES lif_pet.

ENDCLASS.


CLASS lcl_pet IMPLEMENTATION.

ENDCLASS.


CLASS lcl_dog DEFINITION
  INHERITING FROM lcl_pet.

  PUBLIC SECTION.
    METHODS lif_animal~make_sound REDEFINITION.

ENDCLASS.


CLASS lcl_dog IMPLEMENTATION.
  METHOD lif_animal~make_sound.
    WRITE `Implementation is here, we're fine`.
  ENDMETHOD.
ENDCLASS.
```

**Comment by fdupret** [comment]
I'll try to fix it
